Pinta.Tools/Editable/EditEngines/BaseEditEngine.cs

```diff
@@ -457,186 +459,223 @@ public virtual void HandleAfterRedo ()
 	public virtual bool HandleKeyDown (Document document, ToolKeyEventArgs e)
 	{
 		Gdk.Key keyPressed = e.Key;
+		switch (keyPressed) {
+			case Gdk.Key.Delete:
+				HandleDelete ();
+				return true;
+			case Gdk.Key.Return:
+			case Gdk.Key.KP_Enter:
+				//Finalize every editable shape not yet finalized.
+				FinalizeAllShapes ();
+				return true;
+			case Gdk.Key.space:
+				HandleSpace (e);
+				return true;
+			case Gdk.Key.Up:
+				//Make sure a control point is selected.
```

Comment:
These comments should go into the extracted methods since they go along with the `SelectedPointIndex` checks